Building package eigen3 failed

> -- CURRENT_INSTALLED_DIR=C:/RnD/vcpkg/installed/x64-windows
-- DOWNLOADS=C:/RnD/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/RnD/vcpkg/packages/eigen3_x64-windows
-- CURRENT_BUILDTREES_DIR=C:/RnD/vcpkg/buildtrees/eigen3
-- CURRENT_PORT_DIR=C:/RnD/vcpkg/ports/eigen3/.
-- Using cached C:/RnD/vcpkg/downloads/eigen-3.3.3.tar.bz2
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Configuring x64-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
    Command failed: C:/RnD/cmake/bin/cmake.exe;C:/RnD/vcpkg/buildtrees/eigen3/src/eigen-eigen-67e894c6cd8f;-DBUILD_SHARED_LIBS=ON;-DVCPKG_TARGET_TRIPLET=x64-windows;-DCMAKE_CXX_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP ;-DCMAKE_C_FLAGS= /DWIN32 /D_WINDOWS /W3 /utf-8 /MP ;-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON;-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON;-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE;-DCMAKE_VERBOSE_MAKEFILE=ON;-DVCPKG_APPLOCAL_DEPS=OFF;-DCMAKE_TOOLCHAIN_FILE=C:/RnD/vcpkg/scripts/buildsystems/vcpkg.cmake;-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON;-DCMAKEPACKAGE_INSTALL_DIR=C:/RnD/vcpkg/packages/eigen3_x64-windows/share/eigen3;-DCMAKE_CXX_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi ;-DCMAKE_C_FLAGS_RELEASE=/MD /O2 /Oi /Gy /DNDEBUG /Zi ;-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ;-DCMAKE_EXE_LINKER_FLAGS_RELEASE=/DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ;-G;Ninja;-DCMAKE_BUILD_TYPE=Release;-DCMAKE_INSTALL_PREFIX=C:/RnD/vcpkg/packages/eigen3_x64-windows
    Working Directory: C:/RnD/vcpkg/buildtrees/eigen3/x64-windows-rel
    See logs for more information:
      C:\RnD\vcpkg\buildtrees\eigen3\config-x64-windows-rel-out.log
      C:\RnD\vcpkg\buildtrees\eigen3\config-x64-windows-rel-err.log
Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:170 (vcpkg_execute_required_process)
  ports/eigen3/portfile.cmake:11 (vcpkg_configure_cmake)
  scripts/ports.cmake:73 (include)

Is it really necessary to build a header only library?

[config-x64-windows-rel-err.zip](https://github.com/Microsoft/vcpkg/files/1112185/config-x64-windows-rel-err.zip)

--- a/ports/eigen3/CONTROL
+++ b/ports/eigen3/CONTROL
@@ -1,3 +1,3 @@
 Source: eigen3
-Version: 3.3.3-4
+Version: 3.3.3-5
 Description: C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.

--- a/ports/eigen3/CONTROL
+++ b/ports/eigen3/CONTROL
@@ -1,3 +1,3 @@
 Source: eigen3
-Version: 3.3.3-3
+Version: 3.3.3-4
 Description: C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.

--- a/ports/eigen3/portfile.cmake
+++ b/ports/eigen3/portfile.cmake
@@ -10,7 +10,7 @@ vcpkg_download_distfile(ARCHIVE
 vcpkg_extract_source_archive(${ARCHIVE})
 # check if required file exists
 if((NOT EXISTS ${SOURCE_PATH}/Eigen/CMakeLists.txt) OR (NOT EXISTS ${SOURCE_PATH}/unsupported/Eigen/CMakeLists.txt))
-    message(STATUS "Missing CMakeLists.txt in cache, remove ${CURRENT_BUILDTREES_DIR} and try again.")
+    message(STATUS "Missing CMakeLists.txt in cache, re-extracting.")
     file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}) 
     vcpkg_extract_source_archive(${ARCHIVE})
 endif()
@@ -32,12 +32,9 @@ file(READ "${CURRENT_PACKAGES_DIR}/share/eigen3/Eigen3Targets.cmake" EIGEN_TARGE
 string(REPLACE "set(_IMPORT_PREFIX " "get_filename_component(_IMPORT_PREFIX \"\${CMAKE_CURRENT_LIST_DIR}/../..\" ABSOLUTE) #" EIGEN_TARGETS "${EIGEN_TARGETS}")
 file(WRITE "${CURRENT_PACKAGES_DIR}/share/eigen3/Eigen3Targets.cmake" "${EIGEN_TARGETS}")
 
+file(GLOB INCLUDES ${CURRENT_PACKAGES_DIR}/include/eigen3/*)
 # Copy the eigen header files to conventional location for user-wide MSBuild integration
-file(COPY ${SOURCE_PATH}/Eigen DESTINATION ${CURRENT_PACKAGES_DIR}/include)
-# and no need to leave CMakeLists.txt there
-if(EXISTS ${CURRENT_PACKAGES_DIR}/include/Eigen/CMakeLists.txt)
-    file(REMOVE ${CURRENT_PACKAGES_DIR}/include/Eigen/CMakeLists.txt)
-endif()
+file(COPY ${INCLUDES} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
 
 # Put the licence file where vcpkg expects it
 file(COPY ${SOURCE_PATH}/COPYING.README DESTINATION ${CURRENT_PACKAGES_DIR}/share/eigen3)

--- a/ports/eigen3/portfile.cmake
+++ b/ports/eigen3/portfile.cmake
@@ -6,7 +6,14 @@ vcpkg_download_distfile(ARCHIVE
     FILENAME "eigen-3.3.3.tar.bz2"
     SHA512 bb5a8b761371e516f0a344a7c9f6e369e21c2907c8548227933ca6010fc607a66c8d6ff7c41b1aec3dea7d482ce8c2a09e38ae5c7a2c5b16bdd8007e7a81ecc3
 )
+
 vcpkg_extract_source_archive(${ARCHIVE})
+# check if required file exists
+if((NOT EXISTS ${SOURCE_PATH}/Eigen/CMakeLists.txt) OR (NOT EXISTS ${SOURCE_PATH}/unsupported/Eigen/CMakeLists.txt))
+    message(STATUS "Missing CMakeLists.txt in cache, remove ${CURRENT_BUILDTREES_DIR} and try again.")
+    file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}) 
+    vcpkg_extract_source_archive(${ARCHIVE})
+endif()
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
@@ -24,6 +31,13 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug)
 file(READ "${CURRENT_PACKAGES_DIR}/share/eigen3/Eigen3Targets.cmake" EIGEN_TARGETS)
 string(REPLACE "set(_IMPORT_PREFIX " "get_filename_component(_IMPORT_PREFIX \"\${CMAKE_CURRENT_LIST_DIR}/../..\" ABSOLUTE) #" EIGEN_TARGETS "${EIGEN_TARGETS}")
 file(WRITE "${CURRENT_PACKAGES_DIR}/share/eigen3/Eigen3Targets.cmake" "${EIGEN_TARGETS}")
+
+# Copy the eigen header files to conventional location for user-wide MSBuild integration
+file(COPY ${SOURCE_PATH}/Eigen DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+# and no need to leave CMakeLists.txt there
+if(EXISTS ${CURRENT_PACKAGES_DIR}/include/Eigen/CMakeLists.txt)
+    file(REMOVE ${CURRENT_PACKAGES_DIR}/include/Eigen/CMakeLists.txt)
+endif()
 
 # Put the licence file where vcpkg expects it
 file(COPY ${SOURCE_PATH}/COPYING.README DESTINATION ${CURRENT_PACKAGES_DIR}/share/eigen3)

